Auth/Platform/Auth.Platform.Android/Credentials/OIDCCredentialManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using AeroGear.Mobile.Auth.Credentials;
+using Android.App;
+using Android.Content;
+using Java.Lang;
+
+namespace AeroGear.Mobile.Auth.Credentials
+{
+    /// <summary>
+    /// Class for peristing, retrieving, updating and removing OpenID Connect
+    /// credentials on a device.
+    /// </summary>
+    public class OIDCCredentialManager : ICredentialManager
+    {
+        private static readonly string StoreName = "AeroGear.Mobile.Auth.Credentials";
+
+        private ISharedPreferences SharedPreferences;
+
+        private static OIDCCredentialManager instance;
+
+        /// <summary>
+        /// Instance of the credential manager using the application context.
+        /// </summary>
+        /// <value>Instance of <see cref="OIDCCredentialManager"/></value>
+        public static OIDCCredentialManager Instance
+        {
+            get
+            {
+                if (instance == null)
```

Comment:
I think this could use the ternary conditional operator, might be a bit long though
```
instance = instance == null ? new OIDCCredentialManager(Application.Context.ApplicationContext) : instance;
return instance;
```

Reply:
Tried this out and it's not very readable, so I'm going to leave this the same. But I am going to change the other suggestion as it reads better. Thanks